generators/test/templates/template_test.go

```diff
@@ -0,0 +1,57 @@
+package test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/gruntwork-io/terratest/modules/terraform"
+	"github.com/gruntwork-io/terratest/modules/test-structure"
+)
+
+func TestTerraformSshExample(t *testing.T) {
+	t.Parallel()
+
+	exampleFolder := "./fixture"
+
+	// At the end of the test, run `terraform destroy` to clean up any resources that were created
+	defer test_structure.RunTestStage(t, "teardown", func() {
+		terraformOptions := test_structure.LoadTerraformOptions(t, exampleFolder)
+		terraform.Destroy(t, terraformOptions)
+	})
```

Comment:
You'd better put the teardown at the end of this surrounding function for better readability.

Reply:
Defer is commonly used to simplify functions that perform various clean-up actions, especially in terratest. But if you think it is better to put the teardown at the end of surrounding functions, I will move that.